wcfsetup/install/files/lib/system/language/LanguageFactory.class.php

```diff
@@ -45,119 +47,98 @@ protected function init()
      */
     public function getLanguage(int $languageID): ?Language
     {
-        return $this->cache->getCache()->getLanguage($languageID);
+        return $this->cache->getLanguage($languageID);
     }
 
     /**
      * Returns the preferred language of the current user.
-     *
-     * @param int $languageID
-     * @return  Language
      */
-    public function getUserLanguage($languageID = 0)
+    public function getUserLanguage(?int $languageID = null): Language
     {
         if ($languageID) {
-            $language = $this->getLanguage($languageID);
+            $language = $this->cache->getLanguage($languageID);
             if ($language !== null) {
                 return $language;
             }
         }
 
         $languageID = $this->findPreferredLanguage();
 
-        return $this->getLanguage($languageID);
+        return $this->cache->getLanguage($languageID);
     }
 
     /**
      * Returns the language with the given language code or null if no such
      * language exists.
-     *
-     * @param string $languageCode
-     * @return ?Language
      */
-    public function getLanguageByCode($languageCode)
+    public function getLanguageByCode(string $languageCode): ?Language
     {
-        // called within WCFSetup
-        if (empty($this->cache->getCache()->codes)) {
-            $sql = "SELECT  languageID
+        $language = $this->getLanguageByCode($languageCode);
+        if ($language === null) {
+            // called within WCFSetup
```

Comment:
I don’t think this change is correct. This will now trigger a query whenever this method is called with an unknown code on runtime, regardless of the setup process.